lib/linked_list.rb

```diff
@@ -76,7 +113,21 @@ def length
     # Time Complexity: ?
     # Space Complexity: ?
     def get_at_index(index)
-      raise NotImplementedError
+      return nil if @head.nil?
+
+      if index == 0
+        return @head
+      else
+        current = @head
+        current_index = 0
+
+        while current.next != nil && current_index < index - 1
+          current = current.next
+          current_index += 1
+        end
+
+        return current.data

```

Comment:
This could end up returning the last element if the index is bigger than the length.

```suggestion
        while !current.nil? && current_index < index 
          current = current.next
          current_index += 1
        end
        
        if current_index == index && current
          return current.data
        end

        return nil
```